src/qiboml/operations/differentiation.py

```diff
@@ -49,11 +59,24 @@ def evaluate(self, x: ndarray, encoding, training, decoding, backend, *parameter
         # construct circuit
         circuit = encoding(x) + training
 
-        # compute first gradient part, wrt data
-        gradient = self.gradient_from_data(
-            data=x,
-            backend=backend,
-        )
+        gradient = []
+        if wrt_inputs:
+            # compute first gradient part, wrt data
+            gradient.append(
+                backend.np.reshape(
+                    backend.np.hstack(
+                        self.gradient_wrt_inputs(
+                            x,
+                            encoding,
+                            circuit,
+                            decoding,
+                        )
+                    ),
+                    (decoding.output_shape[-1], x.shape[-1]),
+                )
+            )
+        else:
+            gradient.append(backend.np.zeros((decoding.output_shape[-1], x.shape[-1])))
```

Comment:
I am not sure we can just pad the gradient vector with values or zeros here. 
It is probably safe until the model structure is: encoding -> circuit -> decoding, but as soon as we will have a mix of trainable and non trainable gates we should probably loop over the gates. 
I say this because I guess the position of the gradients in the list is important, while here we are basically padding it at its beginning.